test/unit/providers/simulation-provider.test.ts

```diff
@@ -99,13 +97,99 @@ describe('fallback tenderly simulator', () => {
       async () => false
     );
     const swapRoute = await simulator.simulate(
-      fromAddressMock,
-      swapOptionsMock,
-      swapRouteMock,
-      amountMock,
-      quoteMock
+      fromAddress,
+      swapOptions,
+      swaproute,
+      amount,
+      quote
     );
     expect(simulateTxStub.called).toBeFalsy();
     expect(swapRoute.simulationStatus).toEqual(SimulationStatus.Unattempted);
   });
 });
+describe('Eth estimate gas simulator', () => {
+    const chainId = ChainId.MAINNET;
+    const simulator = new EthEstimateGasSimulator(chainId, provider, v2PoolProvider, v3PoolProvider);
+    let ethEstimateGasStub: sinon.SinonStub;
+    let simulateTxStub: sinon.SinonStub;
+
+    const swaproute: SwapRoute = {
+        quote: quote,
+        quoteGasAdjusted: quote,
+        estimatedGasUsed: estimatedGasUsed,
+        estimatedGasUsedQuoteToken: quote,
+        estimatedGasUsedUSD: quote,
+        gasPriceWei: estimatedGasUsed,
+        trade: trade,
+        route: route,
+        blockNumber: blockNumber,
+        simulationStatus: SimulationStatus.Succeeded,
+    };
+
+    beforeEach(() => {
+        simulateTxStub = sinon.stub(simulator, <any>'simulateTransaction');
+        simulateTxStub.resolves(swaproute);
+        ethEstimateGasStub = sinon.stub(simulator, <any>'ethEstimateGas');
+        ethEstimateGasStub.resolves(swaproute)
+    });
+    
+    afterEach(() => {
+        sinon.restore();
+    });
+    test('simulates when user has sufficient balance and token is approved', async () => {
+        sinon.stub(simulator, <any>'userHasSufficientBalance').resolves(true);
+        sinon.stub(
+            simulator,
+            <any>'checkTokenApproved',
+        ).resolves(true);
```

Comment:
Since you've mocked out the function that calls this, this does nothing. Change .resolves to 'false' and the test will still pass.